### ISSUE TITLE
Windows - Unknown USB Device (Device Descriptor Request Failed

### ISSUE THREAD

**Issue by toremaannn** [cnlohr/espusb#47]
Hi

Do I need any driver on a windows 2016 server? working fine on a ubuntu, macOS and windows 7

In device manager I am seeing this: Unknown USB Device (Device Descriptor Request Failed

**Comment by cnlohr** [comment]
I don't believe so.  It should work just fine, considering it follows the normal HID classes.  The fact that it says that is a little concerning like maybe it's operating somewhat differently than Windows is expecting.  You will need to run USBPcap to capture the transaction and see what exactly windows is asking and what the device is saying.

**Comment by blueman007** [comment]
For the poor soul who will inevitably read this post one day: if you have a Device Descriptor error on Windows, it's very likely that your D+ and D- cables are swapped. Don't ask me how I know.


You're welcome.

**Comment by myfingerhurt** [comment]
@blueman007 
Thank you very much, you saved my day!
And for the recorder, the D4 and D5 on NodeMCU are not GPIO5 and GPIO4.

I have also found out that ESP-Mx DevKit aka 2AL3B-ESP-M or ESP-M2 is not working with ESPUSB.
The Wi-Fi web-server also dose not start, terminal with 115200 talks gibberish.

![IMG_1813 - Copy](https://user-images.githubusercontent.com/12622279/112268539-7afae200-8cba-11eb-94a2-acbcedf83d58.png)
